source/extensions/io_socket/user_space/io_handle_impl.cc

```diff
@@ -391,18 +392,16 @@ void PassthroughStateImpl::mergeInto(envoy::config::core::v3::Metadata& metadata
   if (metadata_) {
     metadata.MergeFrom(*metadata_);
   }
-  if (filter_state_objects_) {
-    for (const auto& [name, object] : *filter_state_objects_) {
-      try {
-        filter_state.setData(name, object, StreamInfo::FilterState::StateType::Mutable,
-                             StreamInfo::FilterState::LifeSpan::Connection);
-      } catch (const EnvoyException& e) {
-        ENVOY_LOG(trace, "Failed to set data for '{}': {}", name, e.what());
-      }
+  for (const auto& object : filter_state_objects_) {
+    try {
+      filter_state.setData(object.name_, object.data_, object.state_type_,
+                           StreamInfo::FilterState::LifeSpan::Connection, object.stream_sharing_);
+    } catch (const EnvoyException& e) {
+      ENVOY_LOG(trace, "Failed to set data for '{}': {}", object.name_, e.what());
```

Comment:
Ignoring this error seems unsafe; something important may have been discarded. Are there cases where this is expected to happen? Should this have an `ENVOY_BUG()`?